reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/metrics/GaugeMetric.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.metrics;
+
+import java.util.function.Supplier;
+
+/**
+ * Gauge metric, an instant value like cpu usage, state and so on
+ */
+public class GaugeMetric<T> extends Metric<T>  {
+    private Supplier<T> loadValue;
```

Comment:
nit: final

Reply:
Removed this class, not needed in current implementation.